web_scrapper.py

```diff
@@ -0,0 +1,48 @@
+import requests
+from bs4 import BeautifulSoup
+import json
+
+
+def remove_html_tags(text):
+    """Remove html tags from a string"""
+    import re
+    clean = re.compile('<.*?>')
+    return re.sub(clean, '', text)
+
+
+listing_html = requests.get(
+    "https://www.linkedin.com/jobs/software-engineer-intern-jobs-united-states/")
+
+doc = BeautifulSoup(listing_html.text, "html.parser")
+
+
+jobs_ul = doc.find("ul", {"class": "jobs-search__results-list"})
+
+data_dump = {}
+for li in jobs_ul.find_all("li"):
```

Comment:
Could you make the code ore self-documenting?  For example, here, I am guessing you are looking at a bulleted list in the html and are iterating through each bullet.  But I have no idea why.  What are trying to extract from each bullet item?

Seems that this code should really be housed inside a function, a function that has a self-documenting name (or else, less optimally, a function with a good comment in front of it).  That function should then be called from here.  Moreover, the function should probably have helper functions to make it more self documenting (or else comments above logical code blocks).

Otherwise, right now, it just looks like a wall of code.

Reply:
Changes done:
- added comments
- made code more self documenting 
- organized code into functions